src/pages/iou/request/step/IOURequestStepDistance.js

```diff
@@ -128,20 +144,22 @@ function IOURequestStepDistance({
         Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID));
     }, [report, iouType, reportID, transactionID, backTo]);
 
-    const getError = () => {
+    const getError = useMemo(() => {
         // Get route error if available else show the invalid number of waypoints error.
         if (hasRouteError) {
             return ErrorUtils.getLatestErrorField(transaction, 'route');
         }
+        
+        if (atLeastTwoDifferentWaypointsError) {
+            return {0: 'iou.error.atLeastTwoDifferentWaypoints'};
+        }
 
-        if (_.keys(waypoints).length > 2 && _.size(validatedWaypoints) !== _.keys(waypoints).length) {
+        if (duplicateWaypointsError) {
```

Comment:
Why do you change the order of conditions in `getError` function. In the main branch, we check `duplicateWaypointsErrorMessage` first

Reply:
Ok